corehq/apps/accounting/task_utils.py

```diff
@@ -17,27 +19,37 @@
 from corehq.util.view_utils import absolute_reverse
 
 
-def get_context_to_send_autopay_failed_email(invoice_id):
-    invoice = Invoice.objects.get(id=invoice_id)
-    subscription = invoice.subscription
-    auto_payer = subscription.account.auto_pay_user
+def get_context_to_send_autopay_failed_email(invoice_id, is_customer_invoice=False):
+    if is_customer_invoice:
+        invoice = CustomerInvoice.objects.get(id=invoice_id)
+        subscription_plan = None
+    else:
+        invoice = Invoice.objects.get(id=invoice_id)
+        subscription_plan = invoice.subscription.plan_version.plan.name
+
+    account = invoice.account
+    auto_payer = account.auto_pay_user
     payment_method = StripePaymentMethod.objects.get(web_user=auto_payer)
-    autopay_card = payment_method.get_autopay_card(subscription.account)
-    domain = invoice.get_domain()
+    autopay_card = payment_method.get_autopay_card(account)
+    domain_name = invoice.get_domain()
     web_user = WebUser.get_by_username(auto_payer)
     if web_user:
-        recipient = web_user.get_email() if web_user.is_active_in_domain(domain) else invoice.get_contact_emails()
+        if _user_active_in_domains(web_user, domain_name, account.name):
+            recipient = web_user.get_email()
+        else:
+            recipient = invoice.get_contact_emails()
     else:
         recipient = auto_payer
 
     template_context = {
-        'domain': domain,
-        'subscription_plan': subscription.plan_version.plan.name,
+        'domain_or_account': domain_name or account.name,
         'billing_date': date.today(),
         'invoice_number': invoice.invoice_number,
         'autopay_card': autopay_card,
-        'domain_url': absolute_reverse('dashboard_default', args=[domain]),
-        'billing_info_url': absolute_reverse('domain_update_billing_info', args=[domain]),
+        'is_customer_invoice': is_customer_invoice,
+        'subscription_plan': subscription_plan,
+        'domain_url': absolute_reverse('dashboard_default', args=[domain_name]),
```

Comment:
might be clearer to leave these out if `domain_name` is `None`

Reply:
https://github.com/dimagi/commcare-hq/pull/37083/commits/81cab194e92d9869e8fa55570ee940a15f4c7bf6